frontend/src/components/accountPage/UserList/UserList.tsx

```diff
@@ -1,240 +1,204 @@
+          />
+          <Column
+            responsive={['sm', 'md', 'lg']}
+            title="Email"
+            dataIndex="email"
+            ellipsis={true}
+            key="email"
+            width={150}
+          />
+          <Column
+            responsive={['md', 'lg']}
+            title={<>Role</>}
+            dataIndex="currentRole"
+            sorter={(a: UserAccountPage, b: UserAccountPage) =>
+              a.currentRole?.localeCompare(b.currentRole!) || 0
+            }
+            key="currentRole"
+            width={80}
+          />
+          <Column
+            title="Action"
+            key="x"
+            width={60}
+            render={(_: any, record: UserAccountPage) =>
+              props.users.length >= 1 ? (
+                <div className="flex justify-center">
+                  <Tooltip title="Swap role">
+                    <SwapOutlined
+                      className="mr-2"
+                      onClick={() => handleChangeCurrentRole(record)}
+                    />
+                  </Tooltip>
+
+                  <Popconfirm
+                    title="Sure to delete?"
+                    onConfirm={async () => {
+                      props.onDeleteUser && (await props.onDeleteUser(record));
+                    }}
+                  >
+                    <DeleteOutlined className="text-red-500" />
+                  </Popconfirm>
+                </div>
+              ) : null
+            }
+          />
+        </Table>
+
+        <Row className="flex justify-end mt-4">
+          <Button
+            type="primary"
+            onClick={() => setshowUserListModal(true)}
+            className="m-1"
+          >
+            <PlusOutlined />
+          </Button>
+
+          <Button
+            type="primary"
+            className="m-1"
+            icon={<UploadOutlined />}
+            onClick={() => setShowUploadModal(true)}
+          >
             Add many from CSV
           </Button>
-        </Upload>
-        {fileError && (
-          <Text className="m-2" type="danger">
-            {fileError}
-          </Text>
-        )}
-      </Row>
-      <Modal
-        title="Add new User"
-        visible={showUserListModal}
-        footer={null}
-        onCancel={closeModal}
-      >
-        <UserListForm
-          onAddUser={newUser => {
-            setUsers(users => [...users, newUser]);
-            props.onAddUser([newUser]);
-            closeModal();
-          }}
+        </Row>
+        <Modal
+          destroyOnClose={true}
+          title="Add new User"
+          visible={showUserListModal}
+          footer={null}
           onCancel={closeModal}
-        />
-      </Modal>
+        >
+          <UserListFormLogic
+            onAddUser={async (newUser: UserAccountPage, workspaces: any) =>
+              !!(
+                props.onAddUser &&
+                (await props.onAddUser([newUser], workspaces))
+              )
+            }
```

Comment:
Would move this in a const function, also you can do something like `props.onAddUser?.(params)` to call a function that could be optional